generator/internal/codegen_utils.cc

```diff
@@ -75,24 +75,16 @@ std::string ProtoNameToCppName(absl::string_view proto_name) {
   return "::" + absl::StrReplaceAll(proto_name, {{".", "::"}});
 }
 
-StatusOr<std::vector<std::string>> BuildNamespaces(VarsDictionary const& vars,
-                                                   NamespaceType ns_type) {
-  auto iter = vars.find("product_path");
-  if (iter == vars.end()) {
-    return Status(StatusCode::kNotFound,
-                  "product_path must be present in vars.");
+std::vector<std::string> BuildNamespaces(std::string const& product_path,
+                                         NamespaceType ns_type) {
+  std::string name;
+  std::vector<std::string> v =
+      absl::StrSplit(product_path, '/', absl::SkipEmpty());
+  if (v.size() > 2) {
+    name = absl::StrJoin(v.begin() + 2, v.end(), "_");
+  } else {
+    name = absl::StrJoin(v.begin(), v.end(), "_");
```

Comment:
Ergh.  It looks like I forgot to save the well-reasoned "nit" comment I wrote here, and now it has disappeared.

The short version is that I suggested ...
```
  std::string name = absl::StrJoin(v.begin() + (v.size() > 2 ? 2 : 0), v.end(), '_');
```
and
```
  absl::StrAppend(&name, "_internal");
```